src/TS_HW4/hw4.ts

```diff
@@ -0,0 +1,136 @@
+//  1. Створіть інтерфейс, який описує структуру об'єкта,що представляє калькулятор.
+//  Калькулятор повинен мати методи для виконання арифметичних операцій: додавання, віднімання, множення та ділення.
+//  Потім створіть функцію calculate, яка приймає об'єкт цього типу та виконує операцію і повертає результат.
+
+enum Actions {
+  Addition,
+  Subtraction,
+  Multiplication,
+  Division,
+}
+
+interface ICalculator {
+  addition(firstNumber: number, secondNumber: number): number;
+  subtraction(firstNumber: number, secondNumber: number): number;
+  multiplication(firstNumber: number, secondNumber: number): number;
+  division(firstNumber: number, secondNumber: number): number;
+}
+
+class Calculator implements ICalculator {
+  addition(firstNumber: number, secondNumber: number): number {
+    return firstNumber + secondNumber;
+  }
+
+  subtraction(firstNumber: number, secondNumber: number): number {
+    return firstNumber - secondNumber;
+  }
+
+  multiplication(firstNumber: number, secondNumber: number): number {
+    return firstNumber * secondNumber;
+  }
+
+  division(firstNumber: number, secondNumber: number): number {
+    return firstNumber / secondNumber;
+  }
+}
+
+function calculate(calc: ICalculator, action: Actions, firstNumber: number, secondNumber: number): number {
+  switch (action) {
+    case Actions.Addition:
+      return calc.addition(firstNumber, secondNumber);
+    case Actions.Subtraction:
+      return calc.subtraction(firstNumber, secondNumber);
+    case Actions.Multiplication:
+      return calc.multiplication(firstNumber, secondNumber);
+    case Actions.Division:
+      return calc.division(firstNumber, secondNumber);
+  }
```

Comment:
Just a quick tip for you: if you're using a "switch" construction, it's always a good idea to include a 'default' case to handle any exceptions that may come up. This will help ensure that your code runs smoothly and without any unexpected errors. Give it a try and let me know how it goes!

Reply:
added:
    default:
      throw new Error('Incorrect type of operation');

Thougth, normally the function may takes only operations from enum, so there will be error on compilation, and default case will never reached?

Reply:
Yes, that's correct!

When the "Actions" enum is modified by adding new values, it is a good practice to handle the changes in the default case.